backend/src/crd/controller/scheduledworkflow/util/cron_schedule.go

```diff
@@ -44,16 +44,35 @@ func NewCronSchedule(cronSchedule *swfapi.CronSchedule) *CronSchedule {
 // scheduled.
 func (s *CronSchedule) GetNextScheduledEpoch(lastJobEpoch *int64,
 	defaultStartEpoch int64) int64 {
+	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
+	return s.getNextScheduledEpoch(effectiveLastJobEpoch)
+}
+
+func (s *CronSchedule) GetNextScheduledEpochNoCatchup(lastJobEpoch *int64,
+	defaultStartEpoch int64, nowEpoch int64) int64 {
+
+	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
+	return s.getNextScheduledEpochImp(effectiveLastJobEpoch, false, nowEpoch)
+}
+
+func (s *CronSchedule) getEffectiveLastJobEpoch(lastJobEpoch *int64,
```

Comment:
can we have some comments on the logics on how next scheduled time is determined, depending on whether the last job finishes or not?

Reply:
Sure, I can add some comments.

It doesn't depend on whether last job finishes or not.
The logic here means when there isn't an existing job, then use schedule start time, or swf creation time as start

Reply:
Done